File: common/src/main/java/com/mall/common/entity/Order.java
package com.mall.common.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 订单实体 — order-service 管理
 */
@Data
@TableName("mall_order")
public class Order {
    @TableId(type = IdType.ASSIGN_ID)       // 雪花算法生成分布式ID
    private Long id;
    private String orderNo;                 // 订单号（业务唯一）
    private Long userId;                    // 用户ID
    private Long productId;                 // 商品ID
    private String productName;             // 商品名称（冗余，防止商品修改后订单信息变化）
    private Integer quantity;               // 数量
    private BigDecimal unitPrice;           // 单价
    private BigDecimal totalAmount;         // 总金额
    private String status;                  // PENDING/PAID/CANCELLED/REFUNDED
    private Long addressId;                 // 收货地址ID
    private String receiverName;            // 收货人姓名快照
    private String receiverPhone;           // 收货人电话快照
    private String shippingAddress;         // 完整地址快照
    private LocalDateTime payTime;          // 支付时间
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: frontend/src/stores/order.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import { listOrders, getOrderDetail, cancelOrder as cancelOrderApi } from '@/api/order'

export interface OrderItem {
  id: number; orderNo: string; userId: number
  productId: number; productName: string
  quantity: number; unitPrice: number; totalAmount: number
  status: string; payTime: string; createTime: string
}

export const useOrderStore = defineStore('order', () => {
  const orders = ref<OrderItem[]>([])
  const loading = ref(false)

  async function fetchOrders(page = 1, size = 50) {
    loading.value = true
    try {
      const res = await listOrders(page, size)
      orders.value = res.data?.records || []
    } finally {
      loading.value = false
    }
  }

  async function cancelOrder(id: number) {
    await cancelOrderApi(id)
    // 本地更新状态
    const order = orders.value.find(o => o.id === id)
    if (order) order.status = 'CANCELLED'
  }

  return { orders, loading, fetchOrders, cancelOrder }
})
